src/wallet/wallet.cpp

```diff
@@ -3076,8 +3076,6 @@ bool CWallet::GetNewChangeDestination(const OutputType type, CTxDestination& des
 {
     error.clear();
 
-    m_spk_man->TopUp();
```

Comment:
In commit "Key pool: Move TopUp call" (8b9a1ed9fb5ef64385d3881b1423059e4daaef13)

Sorry if this is a dumb question, because my understanding here is not great, but the PR description says this call has to be moved so "it will work after future changes where m_spk_man is not a global thing in CWallet and is fetched based on type and internal-ness".

But why couldn't the code here just look up the right keyman object using the type above and the internal-ness which it already knows?

Not saying this shouldn't be moved, just that this doesn't seem to explain the move

Reply:
The place it is moved to does the lookup as well. It was redundant to do the lookup twice.

Reply:
Updated the commit message.